core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -278,7 +278,6 @@ abstract class BaseColumnFormatRelation(
       case None => snc.grabLock(table, schemaName, connProperties)
       case Some(a) => null // Do nothing as putInto will release lock
     }
-    if ((lock != null) && lock.isInstanceOf[RegionLock]) lock.asInstanceOf[RegionLock].lock()
```

Comment:
These explicit lock acquire were redundant and grabLock itself takes the lock?

Reply:
Just moved the code inside the grablock.